.github/workflows/master-push.yml

```diff
@@ -0,0 +1,45 @@
+name: 'Master Push'
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+
+  nix-cache:
+    name: 'Populate Nix Cache'
+    strategy:
+      matrix:
+        runner: [normal, macos-13, ARM64]
+    runs-on: ${{ matrix.runner }}
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.push.head.sha }}
+          fetch-depth: 0
+      - name: 'Install Nix'
+        if: ${{ matrix.runner == 'macos-13' }}
+        uses: cachix/install-nix-action@v19
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.13.3/install
+          extra_nix_config: |
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
```

Comment:
@F-WRunTime, please add this to `secrets`.

The job is based on: https://github.com/runtimeverification/evm-semantics/blob/65ba2987ef8aa2caca7001ba03b725e6da514e35/.github/workflows/master-push.yml#L12

Reply:
GITHUB_TOKEN is generated at runtime.